shared/libraries/chassis.cc

```diff
@@ -26,7 +26,7 @@
 
 namespace control {
 
-Chassis::Chassis(const chassis_t chassis) {
+Chassis::Chassis(const chassis_t chassis) : pids_() {
```

Comment:
What is this thing? Do you pass in a nullptr for pids_?

Reply:
This is invoking the default constructor of `PIDController`, which sets Kp, Ki, Kd to all zeros. I reinit them afterwards.
It is formatted like this due to the constraint that `PIDController`s cannot be allocated using `new`.